src/TypeHandler/Asset/AssetHandler.php

```diff
@@ -95,21 +97,53 @@ protected function findOrCreateFile($id, $class = File::class)
      * @param string $url
      * @param string $name
      * @param string|DataObject $class
+     * @param string $transformation
      *
      * @return File|bool
      * @throws \Exception
      */
-    protected function updateFile($id, $updatedAt, $url, $name, $class = File::class)
+    protected function updateFile($id, $updatedAt, $url, $name, $class = File::class, $transformation = '')
     {
         $file = $this->findOrCreateFile($id, $class);
         if ($file->SalsifyUpdatedAt && $file->SalsifyUpdatedAt == $updatedAt) {
-            return $file;
+            if (!$this->isTransformOutOfDate($file, $class, $transformation)) {
+                return $file;
+            }
         }
 
         $file->SalsifyUpdatedAt = $updatedAt;
+        if ($file->hasField('Transformation')) {
+            $file->Transformation = $transformation;
+        }
         $file->setFromStream(fopen($url, 'r'), $name);
 
+        $published = $file->isPublished();
         $file->write();
+
+        if ($published) {
+            $file->publishSingle();
+        }
         return $file;
     }
+
+    /**
+     * @param DataObject $file
+     * @param string $class
+     * @param string $transformation
+     *
+     * @return bool
+     */
+    private function isTransformOutOfDate($file, $class, $transformation)
+    {
+        if (!$file instanceof Image) {
+            return false;
+        }
+
+        /** @var Image|ImageDataExtension $file */
+        if ($file->Transformation == $transformation) {
```

Comment:
Can we bring this down to `return $file->Transformation != $transformation;`?

Reply:
possibly, but normal files don't have the transformation field, just images.

Reply:
Yup, this is just for this transformation logic at the tail end. We'd keep the instance check ,but bring this if block and subsequent returns down to the ternary. I don't think it's necessary required, just noting it.